### PR TITLE
PT-5417 Do not calculate taxes if Enable Tax Calculation is disabled …

### PR DESCRIPTION
**Pull request by Andalexshap**
…in Store settings

### Jira-link:

https://virtocommerce.atlassian.net/browse/PT-5417

### CODE REVIEW

**Review by akak1977 — APPROVED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs [hunk 1]
```diff
@@ -604,14 +609,22 @@ public virtual Task<PromotionResult> EvaluatePromotionsAsync(PromotionEvaluation
...
+
+                if (await _settingsExtensions.GetSettingFromStore<bool>(taxEvalContext.StoreId, StoreSetting.TaxCalculationEnabled))
+                {
+                    result = taxProvider.CalculateRates(taxEvalContext);
+                }
+                else
+                {
+                    result = new List<TaxRate>();
+                }
             }
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs:628]: check Store.TaxEvaluationEnabled setting before getting taxProvider

File: src/XPurchase/VirtoCommerce.XPurchase/Extensions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -24,6 +24,7 @@ public static IServiceCollection AddXPurchase(this IServiceCollection services,
 
             services.AddTransient<ICartValidationContextFactory, CartValidationContextFactory>();
             services.AddTransient<ICartAvailMethodsService, CartAvailMethodsService>();
+            services.AddTransient<SettingsExtensions>();
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/Extensions/ServiceCollectionExtensions.cs:27]: Remove

File: src/XPurchase/VirtoCommerce.XPurchase/Services/CartAvailMethodsService.cs [hunk 3]
```diff
@@ -84,8 +93,14 @@ public async Task<IEnumerable<ShippingRate>> GetAvailableShippingRatesAsync(Cart
             {
                 var taxEvalContext = _mapper.Map<TaxEvaluationContext>(cartAggr);
                 taxEvalContext.Lines.Clear();
-                taxEvalContext.Lines.AddRange(availableShippingRates.SelectMany(x => _mapper.Map<IEnumerable<TaxLine>>(x)));
-                var taxRates = taxProvider.CalculateRates(taxEvalContext);
+                taxEvalContext.Lines.AddRange(availableShippingRates.SelectMany(x => _mapper.Map<IEnumerable<TaxLine>>(x)));
+
+                var taxRates = new List<TaxRate>();
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/CartAvailMethodsService.cs:98]: User cartAggr.Store to get stroe settings
if TaxCalculationEnabled == false there's not need to even get `taxProvider` and `ApplyTaxRates`

File: src/XPurchase/VirtoCommerce.XPurchase/Services/CartAvailMethodsService.cs [hunk 4]
```diff
@@ -129,8 +144,14 @@ public async Task<IEnumerable<PaymentMethod>> GetAvailablePaymentMethodsAsync(Ca
             {
                 var taxEvalContext = _mapper.Map<TaxEvaluationContext>(cartAggr);
                 taxEvalContext.Lines.Clear();
-                taxEvalContext.Lines.AddRange(result.Results.SelectMany(x => _mapper.Map<IEnumerable<TaxLine>>(x)));
-                var taxRates = taxProvider.CalculateRates(taxEvalContext);
+                taxEvalContext.Lines.AddRange(result.Results.SelectMany(x => _mapper.Map<IEnumerable<TaxLine>>(x)));
+
+                var taxRates = new List<TaxRate>();
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/CartAvailMethodsService.cs:149]: User cartAggr.Store to get stroe settings
if TaxCalculationEnabled == false there's not need to even get `taxProvider` and `ApplyTaxRates`

**Review by ksavosteev — CHANGES_REQUESTED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/Services/CartAvailMethodsService.cs [hunk 5]
```diff
@@ -193,11 +203,15 @@ public async Task<IEnumerable<GiftItem>> GetAvailableGiftsAsync(CartAggregate ca
             }).ToList();
         }
 
-        protected async Task<TaxProvider> GetActiveTaxProviderAsync(string storeId)
-        {
+        protected async Task<TaxProvider> GetActiveTaxProviderAsync(CartAggregate cartAggregate)
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/Services/CartAvailMethodsService.cs:206]: Add a new overloaded method instead
make the old one Obsolete

File: src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs [hunk 6]
```diff
@@ -799,10 +805,10 @@ protected virtual void EnsureCartExists()
         protected async Task<TaxProvider> GetActiveTaxProviderAsync()
         {
             //PT-5417: Do not calculate taxes if Enable Tax Calculation is disabled in Store settings
-            //if (!context.StoreTaxCalculationEnabled)
-            //{
-            //    return;
-            //}
+            if (!await _crudStoreService.GetSettingValue<bool>(Cart.StoreId, StoreSetting.TaxCalculationEnabled))
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs:808]: Use Store property

File: src/VirtoCommerce.ExperienceApiModule.Core/Extensions/StoreServiceExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
+
+using System.Threading.Tasks;
+using VirtoCommerce.Platform.Core.GenericCrud;
+using VirtoCommerce.Platform.Core.Settings;
+using VirtoCommerce.StoreModule.Core.Model;
+
+namespace VirtoCommerce.ExperienceApiModule.Core.Extensions
+{
+    public static class StoreServiceExtensions
```
Comment (ksavosteev) [inline src/VirtoCommerce.ExperienceApiModule.Core/Extensions/StoreServiceExtensions.cs:9]: Remove this class

**Review by ksavosteev — COMMENTED** [review]
Inject StoreService into EvalProductsTaxMiddleware and check TaxEnabled setting

**Review by ksavosteev — APPROVED** [review]

**Pull request merged**